.github/workflows/macos.yml

```
@@ -18,6 +25,34 @@ jobs:
       with:
         java-version: 11
         distribution: 'temurin'
+    - name: Build GDAL
+      run: |
+        mkdir $GDAL_INSTALL_PATH
+        brew update
+        brew upgrade
+        brew install freexl libxml2 libspatialite geos proj libgeotiff openjpeg libaec postgis poppler doxygen unixodbc aom jpeg-xl libheif libarchive libkml boost ccache swig python-setuptools
+        : # gdal is automatically installed as a dependency for postgis
+        brew uninstall --ignore-dependencies gdal
+        wget https://github.com/OSGeo/gdal/releases/download/v$OSX_GDAL_VERSION/gdal-$OSX_GDAL_VERSION.tar.gz
+        tar -xvf gdal-$OSX_GDAL_VERSION.tar.gz
+        cd gdal-$OSX_GDAL_VERSION
+        mkdir build
+        cd build
+        : # See https://github.com/OSGeo/gdal/issues/6949 (Java bindings) and https://github.com/OSGeo/gdal/pull/8226 (Arrow and Parquet)
+        cmake -DCMAKE_INSTALL_PREFIX=$GDAL_INSTALL_PATH -DCMAKE_BUILD_TYPE=Release -DBUILD_JAVA_BINDINGS=ON -DBUILD_PYTHON_BINDINGS=OFF ${CMAKE_OPTIONS} ..  
+        cmake --build .
+        cmake --build . --target install
+        ls $GDAL_INSTALL_PATH/lib/
+        ls $GDAL_INSTALL_PATH/lib/jni/
+        ls /usr/lib/
+        otool -L $GDAL_INSTALL_PATH/lib/libgdal.35.dylib
+        otool -l $GDAL_INSTALL_PATH/lib/libgdal.35.dylib
+        otool -L $GDAL_INSTALL_PATH/lib/libgdal.35.$OSX_GDAL_VERSION.dylib
+        otool -l $GDAL_INSTALL_PATH/lib/libgdal.35.$OSX_GDAL_VERSION.dylib
+        otool -L $GDAL_INSTALL_PATH/lib/jni/libgdalalljni.dylib
+        otool -l $GDAL_INSTALL_PATH/lib/jni/libgdalalljni.dylib
+        java --version
+        java -Djava.library.path=$DYLD_LIBRARY_PATH:$GDAL_INSTALL_PATH:/lib/jni/libgdalalljni.dylib -classpath $GDAL_INSTALL_PATH/share/java/gdal-$OSX_GDAL_VERSION.jar:$PWD/swig/java/build/apps gdalinfo
```

isn't there a typo here with the ':' instead of a '/' between ``$GDAL_INSTALL_PATH`` and ``/lib/jni`` ? Furthermore you need to specify a directory name, not a filename. At least that's how it works for me
```suggestion
        java -Djava.library.path=$DYLD_LIBRARY_PATH:$GDAL_INSTALL_PATH/lib/jni -classpath $GDAL_INSTALL_PATH/share/java/gdal-$OSX_GDAL_VERSION.jar:$PWD/swig/java/build/apps gdalinfo
```

---

... And in any case, the bindings are still failing with those changes.

I was planning on reporting this bindings failure as a bug to GDAL this week, once I find a little time to collect everything I've found here and clean it up into a proper report.

In advance of that, if you've got any other suggestions, I'm all ears.